test_ml_pipeline.py

```
@@ -0,0 +1,69 @@
+    context_karnataka = {"location": "Karnataka", "crop": "rice"}
+    
+    print(f"Routing for Punjab: {router.route(context_punjab)}")
+    print(f"Routing for Karnataka/Rice: {router.route(context_karnataka)}")
+    
+    # 5. Test Prediction
+    try:
+        prediction = router.predict(sample_input, context_punjab)
+        print(f"Prediction Result: {prediction}")
+        assert isinstance(prediction, float)
+        print("SUCCESS: Pipeline Test Successful!")
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)